src/Evolution/Systems/RadiationTransport/M1Grey/M1Closure.cpp

```diff
@@ -149,117 +158,118 @@ void M1Closure(
-      // + hh_thin_thin * dThin^2 + hh_thick_thick * dThick^2
-      // + hh_thin_thick * dThin * dThick;
-      const double hh_0 = -square(h_0_n) + square(h_0_v) * v2_pt +
-                          square(h_0_f) * f2_pt +
-                          2. * h_0_v * h_0_f * v_dot_f_pt;
-      const double hh_thin =
-          2. * (h_0_v * h_thin_v * v2_pt + h_0_f * h_thin_f * f2_pt +
+      // H^2 = h_sqr_0 + h_sqr_thin * d_thin + h_sqr_thick*d_thick
+      // + h_sqr_thin_thin * d_thin^2 + h_sqr_thick_thick * d_thick^2
+      // + h_sqr_thin_thick * d_thin * d_thick;
+      const double h_sqr_0 = -square(h_0_n) + square(h_0_v) * v_sqr_pt +
+                             square(h_0_f) * s_sqr_pt +
+                             2. * h_0_v * h_0_f * v_dot_f_pt;
+      const double h_sqr_thin =
+          2. * (h_0_v * h_thin_v * v_sqr_pt + h_0_f * h_thin_f * s_sqr_pt +
                 h_0_v * h_thin_f * v_dot_f_pt + h_0_f * h_thin_v * v_dot_f_pt -
                 h_0_n * h_thin_n);
-      const double hh_thick =
-          2. * (h_0_v * h_thick_v * v2_pt + h_0_f * h_thick_f * f2_pt +
+      const double h_sqr_thick =
+          2. * (h_0_v * h_thick_v * v_sqr_pt + h_0_f * h_thick_f * s_sqr_pt +
                 h_0_v * h_thick_f * v_dot_f_pt +
                 h_0_f * h_thick_v * v_dot_f_pt - h_0_n * h_thick_n);
-      const double hh_thin_thick =
-          2. * (h_thin_v * h_thick_v * v2_pt + h_thin_f * h_thick_f * f2_pt +
-                h_thin_v * h_thick_f * v_dot_f_pt +
-                h_thin_f * h_thick_v * v_dot_f_pt - h_thin_n * h_thick_n);
-      const double hh_thick_thick =
-          square(h_thick_v) * v2_pt + square(h_thick_f) * f2_pt +
+      const double h_sqr_thin_thick =
+          2. *
+          (h_thin_v * h_thick_v * v_sqr_pt + h_thin_f * h_thick_f * s_sqr_pt +
+           h_thin_v * h_thick_f * v_dot_f_pt +
+           h_thin_f * h_thick_v * v_dot_f_pt - h_thin_n * h_thick_n);
+      const double h_sqr_thick_thick =
+          square(h_thick_v) * v_sqr_pt + square(h_thick_f) * s_sqr_pt +
           2. * h_thick_v * h_thick_f * v_dot_f_pt - square(h_thick_n);
-      const double hh_thin_thin =
-          square(h_thin_v) * v2_pt + square(h_thin_f) * f2_pt +
+      const double h_sqr_thin_thin =
+          square(h_thin_v) * v_sqr_pt + square(h_thin_f) * s_sqr_pt +
           2. * h_thin_v * h_thin_f * v_dot_f_pt - square(h_thin_n);
 
-      double& xi = get(*closure_factor)[s];
       // Root finding function
-      const auto XiJSqrMinusH2 = [
-        &e_pt, &j_0, &j_thin, &j_thick, &hh_0, &hh_thick, &hh_thin,
-        &hh_thin_thin, &hh_thick_thick, &hh_thin_thick
-      ](double lxi) noexcept {
-        const double chi = M1ClosureFunction(lxi);
-        const double dchi_dxi = M1ClosureDeriv(lxi);
-        const double dthin = 1.5 * chi - 0.5;
-        const double dthick = 1. - dthin;
-        const double dthin_dxi = 1.5 * dchi_dxi;
-        const double dthick_dxi = -dthin_dxi;
+      const auto XiJSqrMinusHSqr = [
```

Comment:
oops, missed commenting on this before; should be xi_j_sqr_minus_h_sqr

Reply:
No problem, will fix that as well. I still have to work on the documentation before this is ready to be merged anyway.